flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierHandler.java

```diff
@@ -106,13 +121,38 @@ protected void notifyAbortOnCancellationBarrier(long checkpointId) throws IOExce
 	}
 
 	protected void notifyAbort(long checkpointId, CheckpointException cause) throws IOException {
+		resetAlignment();
 		toNotifyOnCheckpoint.abortCheckpointOnBarrier(checkpointId, cause);
 	}
 
 	protected void markCheckpointStart(long checkpointCreationTimestamp) {
 		latestCheckpointStartDelayNanos = 1_000_000 * Math.max(
 			0,
 			System.currentTimeMillis() - checkpointCreationTimestamp);
+
+		resetAlignment();
+		startOfAlignmentTimestamp = System.nanoTime();
+	}
+
+	protected void markAlignmentEnd() {
+		if (startOfAlignmentTimestamp > 0) {
+			markAlignmentEnd(System.nanoTime() - startOfAlignmentTimestamp);
+		}
+		else {
+			markAlignmentEnd(0);
+		}
+	}
+
+	protected void markAlignmentEnd(long alignmentDuration) {
```

Comment:
but `setAlignmentDuration` doesn't capture the fact, that it also finishes the alignment . It might be named `finishAlignmentWithDuration(duration)` but I think it's better to make it consistent with the `markAlignmentEnd()` version. This after all ends the alignment, but it's just bypassing the `System.nanoTime() - startOfAlignmentTimestamp` check.

Reply:
I see your point and would be going with the current solution, but I'd like to add one more point before that:

> but setAlignmentDuration doesn't capture the fact, that it also finishes the alignment .
The tracker never starts an alignment and thus can't finish it. The only plausible method would be `markCheckpointStart`, which shouldn't fiddle with alignment times imho (which would also avoid calling `setAlignmentDuration` as it's 0 by default).

Reply:
According to now newly redefined `alignmentDuration`, `CheckpointBarrierTracker` does start the alignment. Through `markCheckpointStart`. Should I rename the `markCheckpointStart` to `markAlignmentStart`?